### ISSUE TITLE
Support a config file 

### ISSUE THREAD

**Issue by alexharv074** [xorpaul/g10k#70]
It would be great if this tool supported a config file at somewhere like `~/.git/config.yml`. My immediate use-case is that I want to move the `cachedir` on my development laptop only, perhaps to `/var/tmp/g10k` because `/tmp` is cleaned every time the machine is booted. It looks like while I could specify an alternate cache dir on the command line, that's not ideal because I would need to either change it in the code which is also used in the CI system.

**Comment by xorpaul** [comment]
Sure, I'll look into it.

In the meantime you could use an alias on your dev laptop that uses a different `cachedir`

Something like:
```
alias g10k="g10k -cachedir /tmp/foobar"
```

**Comment by jovrum** [comment]
Maybe another potential workaround would be using an overridable environment variable to specify the `cachedir`.

**Comment by xorpaul** [comment]
There's already
```
g10k_cachedir=./tmp
```

, but that is only available in puppetfile mode.

I can also add that to the config mode.

**Comment by alexharv074** [comment]
So, I just run:

```
export g10k_cachedir=/var/tmp
```

And that will be my cachedir when I later run:

```
g10k -puppetfile
```
?

If so, that's perfect.

Maybe mention this awesomeness in the README?

**Comment by xorpaul** [comment]
Yep, that should do what you want, if you're only using the puppetfile mode.

I'm also not sure why this feature hasn't made it to the README :thinking:

**Comment by xorpaul** [comment]
I added a hint to the README.
The environment variable `g10k_cachedir` can now also override the `:cachedir` setting from the g10k YAML config and also takes precedence over the `-cachedir` CLI parameter.

Can you think of another reason a special config file would be helpful or can I close this issue?

**Comment by alexharv074** [comment]
I suppose if there is just one parameter to set, environment variables are fine whereas if you had to set a few of these parameters, then a config file probably would make sense. Right now, an environment variable is perfect for me.
